rpcd/playbooks/roles/rpc_post_upgrade/tasks/post-upgrade-utility.yml

```diff
@@ -65,3 +65,57 @@
     msg: "One or more of the cinder services are in the down state. Check the output above for more detail."
   when: "'down' in '{{ item.split(' ')[-1] }}'"
   with_items: cinder_output.stdout_lines|default([])
+
+- name: Obtain a list of neutron routers
+  shell: |
+    . {{ ansible_env.HOME }}/openrc
+    neutron router-list -f value --column id
+  register: neutron_routers
+
+- name: Check agent assigned to each neutron router
+  shell: |
+    . {{ ansible_env.HOME }}/openrc
+    neutron l3-agent-list-hosting-router -f value --column host {{ item }}
+  with_items: "{{ neutron_routers.stdout_lines|default([]) }}"
+  register: neutron_routers_agents
+
+- name: Write file containing agent for each router
+  copy:
+    content: "{{ item.stdout }}"
+    dest: "/tmp/{{ item.item }}.post-upgrade"
+  with_items: "{{ neutron_routers_agents.results|default([]) }}"
+  when: item.stdout != ""
+  delegate_to: localhost
+
+- name: Get stat of pre-upgrade files
+  stat:
+    path: "/tmp/{{ item }}.pre-upgrade"
+  register: neutron_routers_pre_stat
+  with_items: "{{ neutron_routers.stdout_lines|default([]) }}"
+  delegate_to: localhost
+
+- name: Get stat of post-upgrade files
+  stat:
+    path: "/tmp/{{ item }}.post-upgrade"
+  register: neutron_routers_post_stat
+  with_items: "{{ neutron_routers.stdout_lines|default([]) }}"
+  delegate_to: localhost
+
+- name: Print warning if neutron router has moved
```

Comment:
This would require the engineer to review the logs, the other checks in this file cause a failure if something has changed. While failing on the first error is not very helpful when trying to generate a report of all services, only printing a message might cause this warning to be missed, mixing the way errors are reported is more likely to lead to mistakes.